synapse/handlers/room_member.py

```diff
@@ -564,25 +565,26 @@ async def update_membership(
         # by application services), and then by room ID.
         async with self.member_as_limiter.queue(as_id):
             async with self.member_linearizer.queue(key):
-                result = await self.update_membership_locked(
-                    requester,
-                    target,
-                    room_id,
-                    action,
-                    txn_id=txn_id,
-                    remote_room_hosts=remote_room_hosts,
-                    third_party_signed=third_party_signed,
-                    ratelimit=ratelimit,
-                    content=content,
-                    new_room=new_room,
-                    require_consent=require_consent,
-                    outlier=outlier,
-                    historical=historical,
-                    allow_no_prev_events=allow_no_prev_events,
-                    prev_event_ids=prev_event_ids,
-                    state_event_ids=state_event_ids,
-                    depth=depth,
-                )
+                with opentracing.start_active_span("update_membership_locked"):
```

Comment:
we should probably give the Linearizer some tracing annotations at some point, which would probably make this redundant. but that's a bigger job and this is certainly fine for now.

Reply:
Curious what that job would entail?

Reply:
really, much the same as you're doing here - adding `with opentracing.start_active_span(...)`  calls around things that might be slow. I'd probably add one at the top level of `Linearizer.queue` (or rather, in the nested `_ctx_manager`), and another at the point where we wait for the lock (`await new_defer`).

Reply:
I think this was asked elsewhere but I didn't see the answer- is there any performance penalty or other drawback to consider when adding opentracing spans?

Reply:
Yes, there's a small performance penalty, which is one reason not to go completely crazy with adding spans. 

There are two aspects to the performance penalty. One, each call to `start_active_span` results in a few function calls and python objects. Two, if tracing is actually enabled for the current request then we additionally have to serialize all the trace data and send it over the network to the jaeger server, and obviously the more spans, the more work this is.

Plus, in some ways, less can be more when it comes to looking at a trace and understanding what's going on.

Reply:
Makes sense, thanks for explaining!